### PR TITLE
Add helper that abstracts flight/react attachment

### PR DESCRIPTION
**Pull request by jharwig**
- [x] @sfeng88 @rygim @jharwig @joeybrk372
- [x] @srfarley @joeferner
- [x] @EvanOxfeld @mwizeman @dsingley

Needed to support extensions that can load react or legacy flight components.

* Includes an `attacher` helper with tests to abstract flight/react attachment
* Converts admin/pluginList to react
* Improved admin UI with loading indicators and better component cleanup
* Adds `url` to admin plugin extension to open new windows instead of rendering a component

This helper is only meant to be used for using an extension point needing to attach plugin components. Use pure react to attach built-in components.

### CODE REVIEW

File: web/war/src/main/webapp/js/admin/bundled/pluginList/PluginList.jsx [hunk 1]
```diff
@@ -0,0 +1,132 @@
+define([
+    'react'
+], function(React) {
+    'use strict';
+
+    const DetailKeys = 'fileName className projectVersion builtBy builtOn gitRevision'.split(' ');
+    const TypeDateKeys = 'builtOn'.split(' ');
+    const PropTypes = React.PropTypes;
+    const PluginItem = React.createClass({
```
Comment (joeferner) [inline web/war/src/main/webapp/js/admin/bundled/pluginList/PluginList.jsx]: Should `PluginItem` and `PluginSection` be in separate files?

File: web/war/src/main/webapp/js/admin/bundled/pluginList/PluginList.jsx [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+                        return !!item.value;
+                    })
+
+            return (
+                <li>
+                    <h1 className="name">{item.name}</h1>
+                    <h2 className="description">{item.description}</h2>
+                    <dl>
+                    {details.map(function(i) {
+                        return [<dt className={i.key}>{i.display}</dt>, <dd className={i.key}>{i.value}</dd>]
```
Comment (joeferner) [inline web/war/src/main/webapp/js/admin/bundled/pluginList/PluginList.jsx]: Are these `className`s being used?

**Comment by joeferner** [comment]
I like this, having to put a connect in every react component was a pain.

File: web/war/src/main/webapp/js/admin/bundled/pluginList/PluginList.jsx (hunk 1 above)
Comment (EvanOxfeld) [inline web/war/src/main/webapp/js/admin/bundled/pluginList/PluginList.jsx]: Yes that's preferred -- https://github.com/v5analytics/visallo/tree/master/web/war/src/main/webapp#basic-rules-1 @jharwig Is there a reason you included multiple components in one file?

Comment (jharwig) [inline web/war/src/main/webapp/js/admin/bundled/pluginList/PluginList.jsx]: Not sure, these are internal components for the list that wouldn't be used elsewhere. Until webpack, there are performance advantages here.

File: web/war/src/main/webapp/js/admin/bundled/pluginList/PluginList.jsx (hunk 2 above)
Comment (jharwig) [inline web/war/src/main/webapp/js/admin/bundled/pluginList/PluginList.jsx]: Right, no longer needed

File: web/war/src/main/webapp/js/admin/bundled/pluginList/PluginList.jsx [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+                name = formatKeyForDisplay(this.props.name);
+            if (this.props.items.length === 0) {
+                sectionClassName += ' disabled'
+            }
+            if (this.state.expanded) {
+                sectionClassName += ' expanded';
+            }
+
+            return (
+                <section onClick={this.toggleCollapsed} className={sectionClassName}>
```
Comment (EvanOxfeld) [inline web/war/src/main/webapp/js/admin/bundled/pluginList/PluginList.jsx]: While string concatenation works, particularly in a fairly simple component like `PluginSection`, I find the [classnames](https://www.npmjs.com/package/classnames) package helpful:

 ```
 <section className={
  classNames('collapsible has-badge-number', {
    disabled: this.props.items.length === 0,
    expanded: this.state.expanded
  })}>
```

**Comment by EvanOxfeld** [comment]
@jharwig Looks great, and the tests were incredibly helpful. 👍 ✨

File: web/war/src/main/webapp/js/admin/bundled/pluginList/PluginList.jsx (hunk 3 above)
Comment (jharwig) [inline web/war/src/main/webapp/js/admin/bundled/pluginList/PluginList.jsx]: Yeah I like that

**Pull request merged**